server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -196,7 +196,7 @@ MappedFieldType failIfFieldMappingNotFound(String name, MappedFieldType fieldMap
         if (fieldMapping != null || allowUnmappedFields) {
             return fieldMapping;
         } else if (mapUnmappedFieldAsString) {
-            TextFieldMapper.Builder builder = new TextFieldMapper.Builder(name, getIndexAnalyzers());
+            TextFieldMapper.Builder builder = new TextFieldMapper.Builder(name, getIndexAnalyzers(), false);
```

Comment:
I think `indexSettings.getMode().isSyntheticSourceEnabled()` can be used here instead?

Reply:
`indexSettings` can be null, there is a constructor that explicitly sets it to null. I was "inspired" by the line below that always sets `MapperBuilderContext#isSourceSynthetic` to `false` in this context. I'll check index settings and fallback to false.